### PR TITLE
Fix SFT for VLM example

### LINKED ISSUE #1786: Lora seems to be invalid when using vsft_llava.py
I used two A100 40g to fine-tune llava-7b with lora. When I used the lora vsft command you provided, I found that the error CUDA out of memory still appeared, so it seems that lora did not work.
My command is as follows, in which I have modified the dataset path:
```
python examples/scripts/vsft_llava.py \
    --dataset_name="../subset/aug_llava_instruct_mix_vsft" \    
    --model_name_or_path="llava-hf/llava-1.5-7b-hf" \
    --report_to="wandb" \
    --learning_rate=1.4e-5 \
    --per_device_train_batch_size=8 \
    --gradient_accumulation_steps=1 \
    --output_dir="../logs/checkpoints/aug-vsft-llava-1.5-7b-hf" \
    --logging_steps=5 \
    --num_train_epochs=1 \
    --push_to_hub \
    --gradient_checkpointing \
    --remove_unused_columns=False \
    --torch_dtype=float16 \
    --fp16=True \ 
    --use_peft=True \
    --lora_r=64 \
    --lora_alpha=16 \
    --lora_target_modules="all-linear"

### PR DESCRIPTION
**Pull request by qgallouedec**
Closes #1786

For some reason, when running sft with llava 1.5 without PEFT, you get this annoying errors:
- loss explosion because of high grad norm in the visual encoder; you need to specify a low `max_grad_norm` (0.1 at least)
- when training with multiple GPUs, you get OOM error (even with batch size = 1). Not sure where it comes from

Consequently,
- I removed the example script without peft
- fixed the command args, in particular the use of bf16

TODO:
- [x] Add a section in SFT documentation with link to run, command, a few remarks, and evaluations results

### CODE REVIEW

**Review by kashif — APPROVED** [review]

**Comment by qgallouedec** [comment]
CI breakes with  transformers-4.43.1 
(passes with 4.39.2)

File: examples/scripts/vsft_llava.py [hunk 1]
```diff
@@ -172,13 +154,13 @@ def __call__(self, examples):
...
+            data_collator=collate_fn,
             train_dataset=train_dataset,
             eval_dataset=eval_dataset,
-            dataset_text_field="text",  # need a dummy field
-            tokenizer=tokenizer,
+            tokenizer=processor.tokenizer,
             peft_config=get_peft_config(model_config),
             callbacks=[RichProgressCallback] if TRL_USE_RICH else None,
-            data_collator=data_collator,
+            dataset_text_field="",  # need a dummy field
```
Comment (kashif) [inline examples/scripts/vsft_llava.py:163]: i think all these options can go into the `SFTConfig` now

Comment (qgallouedec) [reply examples/scripts/vsft_llava.py]: done, thanks!

**Comment by qgallouedec** [comment]
I'm still not sure how to evaluate, given that open-compass/VLMEvalKit is not compatible with llava 1.5. I'll do this later.

**Pull request merged**